core/tx.go

```diff
@@ -260,3 +271,79 @@ func (db *DB) FetchMPWithID(id uint64, accountMP *StateMerkleProof) (err error)
 	*accountMP = accMP
 	return nil
 }
+
+// Validate creates proofs for validating txs and returns new root post validation
+func (tx *Tx) Validate(bz Bazooka, currentRoot ByteArray) (newRoot ByteArray, err error) {
+	fromStateProof, toStateProof, txDBConn, err := tx.GetVerificationData()
+	if err != nil {
+		return
+	}
+
+	newRoot, err = bz.ProcessTx(currentRoot, *tx, fromStateProof, toStateProof)
+	if err != nil {
+		if txDBConn.Instance != nil {
+			txDBConn.Instance.Rollback()
+			txDBConn.Close()
+		}
+		return
+	}
+	if txDBConn.Instance != nil {
+		txDBConn.Instance.Commit()
+		txDBConn.Close()
+	}
+
+	return
+}
+
+// ProcessTxs processes all trasnactions and returns the commitment list
+func ProcessTxs(db DB, bz Bazooka, txs []Tx, isSyncing bool) (commitments []Commitment, err error) {
+	if len(txs) == 0 {
+		return commitments, ErrNoTxsFound
+	}
+	for i, tx := range txs {
+		rootAcc, err := db.GetRoot()
+		if err != nil {
+			return commitments, err
+		}
+		currentRoot, err := HexToByteArray(rootAcc.Hash)
+		if err != nil {
+			return commitments, err
+		}
+		newRoot, err := tx.Validate(LoadedBazooka, currentRoot)
+		if err != nil {
+			return commitments, err
+		}
+		if i%COMMITMENT_SIZE == 0 {
+			txInCommitment := txs[i : i+COMMITMENT_SIZE]
+			aggregatedSig, err := aggregateSignatures(txInCommitment)
+			if err != nil {
+				if isSyncing && err == ErrSignatureNotPresent {
+					continue
```

Comment:
Does the `continue` here mean the line `commitments = append(commitments, commitment)` is skipped?

Reply:
No it goes there, it simply means that if we are syncing and tx.Sig is nil we don't have to bubble up error as that is expected. I do think this piece is smelly and should be refactored in the future.